### DISCUSSION TITLE
Database migration issue is arising when we are trying to use an upgraded version of rudder-server on our self hosted plane.

### DISCUSSION THREAD

**Discussion by Saup21** [rudderlabs/rudder-server#4639] [General]
**Bug Description**

Database migration issue is arising when we are trying to use an upgraded version of rudder-server on our self hosted plane. We were using an older version of rudder-server and thought of upgrading it to the latest versions. After successfully upgrading we started the server by connecting to the old db that we had since we saw proper migration files. But the issue occurred in a way that the workspace_id column was not getting added in the db and prior to that indexing of columns were taking place. This is happening because we are trying to create a new read for errDBForRead before writing. What is happening in turn is that we are not migrating the dbs before indexing.

**Steps to reproduce the bug**
1. Use an older version of the service
2. Create a db with that version
3. Upgrade the service to a newer version but use the existing database
4. Start the service by running `go run`

**Expected behavior**

Successful migration of the old database after which service starts normally.

**Screenshots**

![Screenshot 2023-08-08 at 3 01 12 PM](https://github.com/rudderlabs/rudder-server/assets/75141372/f98207d6-ce98-4903-ab54-345d43523fa3)

**Desktop (please complete the following information):**
 - OS: MacOS - 13.4.1(Ventura)
 - Browser - Chrome
 - Version - 114.0.5735.198

**Comment by Saup21** [comment]
I have full context about the issue and have solved it in my local as well. Please review the issue and assign me the same.

**Comment by gitcommitshow** [comment]
Hi @Saup21 , thank you for pitching in to contribute and apologies for the delayed response. Did you create a PR for this? 
P.S. If you face any delays in response here on GitHub, please use #contributing-to-rudderstack channel in our [slack community](https://www.rudderstack.com/join-rudderstack-slack-community/) to remind about any pending issue/PR. That is a dedicated channel to get support/guidance for the project contributors.
